### PR TITLE
Update fieldset component to follow conventions

### PR DESCRIPTION
**Pull request by igloosi**
- provide text and html variations for the legend and hint parameters,
so developers can use either and in the template we default to text and
mark html as safe
- update YAML file to reflect new parameters
- update table of arguments to reflect changes
- update example in fieldset.njk file
- add functionality to accept an object of additional html attributes

Trello ticket: https://trello.com/c/9iL8IIzc/278-update-fieldset-component-api

### CODE REVIEW

**Comment by hannalaakso** [comment]
Should there be a `classes` attribute in the component? Just noticed a small typo: `with-html-insted-of-text` should be `with-html-instead-of-text`

A more general note: What if you want to place a whole component like an input inside the fieldset? Would it better just to have fieldset as copy-paste markup for flexibility?

File: src/components/fieldset/fieldset.yaml [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
   data:
-    classes: ''
-    legendText: 'Legend text goes here'
-    hintText: 'Legend hint text goes here'
-    errorMessage: 'Error message goes here'
+    legendText: Legend text goes here
+    hintText: Legend hint text goes here
+    errorMessage:
+      text: 'Error message goes here'
+- name: with-html-insted-of-text
```
Comment (hannalaakso) [inline src/components/fieldset/fieldset.yaml]: Should be `with-html-instead-of-text`

File: src/components/fieldset/fieldset.njk [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
-  classes: '',
-  legendText: 'Legend text goes here'
-  })
--}}
+{{ govukFieldset({
+  "legendText": "Legend text goes here",
+  "hintText": "Legend hint text goes here",
+  "errorMessage": {
+    "text": "Error message goes here"
+  }
```
Comment (hannalaakso) [inline src/components/fieldset/fieldset.njk]: Should there be a `classes` attribute?

**Comment by igloosi** [comment]
you can specify classes, it's documented.
you can currently place a component inside the fieldset, as it's been done with date. there is a trello ticket to revisit this type of component (vs guidance around cop&paste markup)
I have amended the typo

**Pull request merged**